build_dmks.sh

```diff
@@ -0,0 +1,125 @@
+#!/bin/bash
+set -e
+
+PROJECT_DIR=${2:-$PWD}
+DRIVER_CODE_DIR=${1:-"$PROJECT_DIR/driver"}
+
+echo "==> Project directory: $PROJECT_DIR"
+echo "==> Driver code directory: $DRIVER_CODE_DIR"
+
+echo "==> Checking for the module name and version in $DRIVER_CODE_DIR/dkms.conf"
+# Define module name and version based on dkms.conf
+MODULE_NAME=$(grep '^PACKAGE_NAME' "$DRIVER_CODE_DIR/dkms.conf" | awk -F'=' '{print $2}' | tr -d ' ')
+VERSION=$(grep '^PACKAGE_VERSION' "$DRIVER_CODE_DIR/dkms.conf" | awk -F'=' '{print $2}' | tr -d ' ')
+
+if [ -z "$MODULE_NAME" ] || [ -z "$VERSION" ]; then
+    echo "Error: Module name or version not found in dkms.conf"
+    exit 1
+fi
+
+echo "==> Module name: $MODULE_NAME"
+echo "==> Version: $VERSION"
+
+echo "==> Checking for required tools and dependencies..."
+# Check if dkms is installed, install if missing
+if ! command -v dkms &> /dev/null; then
+    echo "dkms command not found. Installing dkms..."
+    sudo apt update && sudo apt install -y dkms
+fi
+
+# Check if debhelper is installed, install if missing
+if ! dpkg-query -W -f='${Status}' debhelper 2>/dev/null | grep -q "install ok installed"; then
+    echo "debhelper not found. Installing debhelper..."
+    sudo apt install -y debhelper
+fi
+
+echo "==> Preparing to use dkms to package the module..."
+# Ensure the module directory exists in /usr/src
+WORKSPACE="/usr/src/${MODULE_NAME}-${VERSION}"
+sudo rm -rf "$WORKSPACE"  # Remove if it exists
+sudo mkdir -p $WORKSPACE
+sudo cp -r "$DRIVER_CODE_DIR"/* "$WORKSPACE"
+
+
+# Add the module
+if dkms status -m "$MODULE_NAME" -v "$VERSION" | grep -q "$VERSION"; then
+    echo "Module '$MODULE_NAME' already exists"
+else
+    sudo dkms add -m "$MODULE_NAME" -v "$VERSION"
+fi
+
+# Create Debian package
+sudo dkms mkdeb -m "$MODULE_NAME" -v "$VERSION" --kernelsourcedir="$DKMS_KERNEL_SOURCE_DIR"
+
+
+#(Optional): Create artifact package
```

Comment:
In what sense is this optional? It looks like it all gets executed every time

Reply:
2903a9b354b66d44d5990196de8f75c33bcd6e54